core/src/context.rs

```diff
@@ -22,6 +31,7 @@ pub struct Context {
     state: Option<Arc<RwLock<State>>>,
     pub action_channel: SyncSender<ActionWrapper>,
     pub observer_channel: SyncSender<Observer>,
+    pub file_storage: FilesystemStorage,
```

Comment:
```suggestion
    pub file_storage: Box<dyn ContentAddressableStorage>,
```

Reply:
This needs to be done in another story because it means Context<T>. This causes an implosion of types which need to be changed across the system. This might also mean lifetimes but good suggestion!

Reply:
@lucksus @AshantiMutinta we deliberately said that we would just hardcode for this PR to avoid the type implosion or lifetimes

i was actually kind of assuming a very lightweight actor to return the storage would avoid the type issues, but this is a future conversation

Reply:
I don't understand what you mean with explosion of types and lifetimes.. 

I was saying hardcoding the path for the filestorage the gets set in the context when passing it to core, which means the context does depend on a specific CAS implementation.

The SoA says: "The storage implementation is passed into HC core via the Context."

Reply:
Passing a CAS into the context will mean that the struct has to be refactored to have a type definition e.g. struct context<T:CAS>{...}then you have your Box<T> in that. The reason being the type is valid(so it can be used in methods and return types but can't be constructed in a struct. Context touches so many things so you will likely have to touch every part that uses context that is if you don't pass it around an associated type wrapper

Hardcoding for now gets passed this and attempts and uses it as if it wasn't even a trait. Benefit I could see from doing it this right now is that it makes the refactor much easier only in the chainstore context cause not a lot of things will change. Context taking in a filesystemeav or filesystemcas could potentially be a huge refactor